### ISSUE TITLE
No AES-NI support on FreeBSD 64-bit

### ISSUE THREAD

**Issue by Sp1l** [libressl-portable/portable#135]
The default build currently does not support AES-NI. The problem is that the CPU type on FreeBSD is amd64 where the configure script tests for x86_64.

With a slight modification it works as expected.
type             16 bytes     64 bytes    256 bytes   1024 bytes   8192 bytes
aes-128-cbc     645544.55k   732261.89k   749218.42k   751550.85k   744527.19k
aes-128 cbc      79150.01k    90466.71k    90951.80k    91684.86k    92187.31k

```
--- configure.orig      2015-09-13 23:53:29 UTC
+++ configure
@@ -13258,7 +13258,7 @@ fi


 # Conditionally enable assembly by default
- if test "x$HOST_ABI" = "xelf" -a "$host_cpu" = "x86_64" -a "x$enable_asm" != "xno"; then
+ if test "x$HOST_ABI" = "xelf" -a "$host_cpu" = "amd64" -a "x$enable_asm" != "xno"; then
   HOST_ASM_ELF_X86_64_TRUE=
   HOST_ASM_ELF_X86_64_FALSE='#'
 else
```

**Comment by Sp1l** [comment]
Came up with a patch for configure.ac but this seems to have some side-effects
```
--- configure.ac.orig   2015-09-13 23:53:14 UTC
+++ configure.ac
@@ -104,7 +104,8 @@ AM_CONDITIONAL([OPENSSL_NO_ASM], [test "

 # Conditionally enable assembly by default
 AM_CONDITIONAL([HOST_ASM_ELF_X86_64],
-    [test "x$HOST_ABI" = "xelf" -a "$host_cpu" = "x86_64" -a "x$enable_asm" != "xno"])
+    [test "x$HOST_ABI" = "xelf" -a "$host_cpu" = "x86_64" -a "x$enable_asm" != "xno" ||
+     test "x$HOST_ABI" = "xelf" -a "$host_cpu" = "amd64" -a "x$enable_asm" != "xno" ])
 AM_CONDITIONAL([HOST_ASM_MACOSX_X86_64],
     [test "x$HOST_ABI" = "xmacosx" -a "$host_cpu" = "x86_64" -a "x$enable_asm" != "xno"])
```
Seems to now misplace the .pc files in `lib/pkgconfig` rather than `libdata/pkgconfig`...

**Comment by busterb** [comment]
We may need a more sophisticated test than this, since dual arch platforms
cannot be as easily detected by using a cpu name alone. That is, a test
that builds something and sets an autoconf variable that we use here.

**Comment by Sp1l** [comment]
For now I'm fixing this by modifying the --build arg when the architecture is amd64 on FreeBSD.

**Comment by busterb** [comment]
I think this is resolved for now with 3ac78a5573fd33a5efc7e590856efd38e9c11b1d

Can you please verify on your end as well?

**Comment by Sp1l** [comment]
Confirmed that it's now building assembly.
